File: cplus/leetcode/202509/algo1016.cpp
#include <bits/stdc++.h>
using namespace std;

// 滑动窗口与双指针 - 定长滑动窗口 1.2 进阶
class Solution {
public:
    bool queryString(string s, int n) {
        // 参考题解 脑筋急转弯
        for (int i=1; i<=n; ++i) {
            auto bin = bitset<32>(i).to_string();
            if (s.find(bin.substr(bin.find('1'))) == string::npos) return false;
        }
        return true;
    }
};

File: python/algo/202510/algo1016.py
# 滑动窗口与双指针 - 定长滑动窗口 1.2 进阶
class Solution:
    def queryString(self, s: str, n: int) -> bool:
        return all(bin(i)[2:] in s for i in range(1, n+1))